app/services/user_updater.rb

```diff
@@ -85,9 +85,15 @@ def update(attributes = {})
 
     save_options = false
 
-    # special handling for theme_key cause we need to bump a sequence number
-    if attributes.key?(:theme_key) && user.user_option.theme_key != attributes[:theme_key]
-      user.user_option.theme_key_seq += 1
+    # special handling for theme_id cause we need to bump a sequence number
+    if attributes.key?(:theme_ids)
+      user_guardian = Guardian.new(user)
```

Comment:
wait why do we need another guardian here if the code above already defines one?

Reply:
The guardian above is for the acting user (e.g. when an admin changes a user's preferences), where as this new one is for the target user.